department-of-veterans-affairs/vets-api
# Add service account config for chatbot to seeds

## Summary

This PR adds a service account config for the chatbot service in the development seeds script.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#65558


## Testing done

- Run the seeds script.
- Verify the service account was added successfully.
- Open the Rails console and generate an assertion.
```ruby
private_key = OpenSSL::PKey::RSA.new(
  File.read('spec/fixtures/sign_in/sample_service_account.pem')
)

payload = {
  'iss' => 'http://localhost:4000',
  'sub' => 'vets.gov.user+0@gmail.com',
  'aud' => 'http://localhost:4000',
  'iat' => Time.now.to_i,
  'exp' => 10.days.from_now.to_i,
  'scopes' => ['http://localhost:3000/v0/map_services/token'],
  'service_account_id' => 'chatbot',
  'jti' => SecureRandom.hex
}

JWT.encode(payload, private_key, 'RS256')
```
- Using the Sign in Service postman collection, obtain a service account token.
  - Paste the generated assertion as the value for the `signed_service_account_assertion` collection variable.
  - Send the `token` request (located in the `Service Account Auth` collection.)
- Verify the endpoint responds with a service account access token.

## Screenshots
Not relevant.


## What areas of the site does it impact?
This does not impact current functionality, but will be used as part of the MAP STS endpoint.

## Acceptance criteria


## Requested Feedback

None.
